File: spp-java/src/main/java/pparser/value/ValuesFactoryImpl.java
package pparser.value;

import java.math.BigDecimal;
import java.util.Collections;
import java.util.List;

import pparser.ValuesFactory;

public class ValuesFactoryImpl implements ValuesFactory {

  @Override
  public String createString(String string) {
    return string.substring(1, string.length() - 1);
  }

  @Override
  public Object createNumber(String tokenImage) {
    return new BigDecimal(tokenImage);
  }

  /**
   * Creates a {@link Path} instance for the given routes
   */
  @Override
  public Object createPath(List<String> tokenImages) {
    return new Path(Collections.unmodifiableList(tokenImages));
  }

}

File: spp-scala/src/main/scala/pparser/value/ValuesFactoryImpl.scala
package pparser.value
import pparser.ast._
import pparser.ValuesFactory
import scala.collection.JavaConversions._
object ValuesFactoryImpl extends ValuesFactory {
  
  override def createString(tokenImage : String) = tokenImage.substring(1, tokenImage.length() - 1)
  
  override def createNumber(tokenImage : String)  = BigDecimal(tokenImage) 

  override def createPath(tokenImages : java.util.List[String]) = Var(tokenImages : _*)

}